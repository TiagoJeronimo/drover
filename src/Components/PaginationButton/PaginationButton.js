import React from 'react'
import './PaginationButton.style.scss'

const PaginationButton = ({pageNumber, active, handleChangePage}) => (
    <button 
        className={`paginationButton-button ${active ? 'paginationButton-button--active' : ''}`}
        onClick={() => handleChangePage(pageNumber)}
    >
        {pageNumber}
    </button>
)

export default PaginationButton