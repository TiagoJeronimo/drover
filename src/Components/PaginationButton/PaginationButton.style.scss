
.paginationButton-button {
    cursor: pointer;
    text-decoration: none;
    outline: none;
    font-weight: bold;
    height: 30px;
    width: 30px;
    color: black;
    border-width: 1px;
    border-style: solid;
    border-color: black;
    margin: 0 1px;
    &--active {
        color: white;
        background: rgb(0, 30, 255);
    }
}