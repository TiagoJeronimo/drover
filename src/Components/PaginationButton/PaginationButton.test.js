import React from 'react';
import { mount } from 'enzyme';

import PaginationButton from './PaginationButton';

it('PaginationButton with not active button', () => {
    const props = {
        pageNumber: 1,
        active: false,
        handleChangePage: jest.fn(),
    };
    const component = (
        <PaginationButton {...props} />
    );
    const html = mount(component).html();
    expect(html).toMatchSnapshot();
  });

  it('PaginationButton with active button', () => {
    const props = {
        pageNumber: 2,
        active: true,
        handleChangePage: jest.fn(),
    };
    const component = (
        <PaginationButton {...props} />
    );
    const html = mount(component).html();
    expect(html).toMatchSnapshot();
  });


