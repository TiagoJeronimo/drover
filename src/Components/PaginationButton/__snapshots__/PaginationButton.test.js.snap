// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationButton with active button 1`] = `"<button class=\\"paginationButton-button paginationButton-button--active\\">2</button>"`;

exports[`PaginationButton with not active button 1`] = `"<button class=\\"paginationButton-button \\">1</button>"`;
